mautic/mautic
# Refactor and small optimization for dynamic content html (part 2/2)

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | yes/no
| New feature?                           | no
| Deprecations?                          | yes
| BC breaks?                             | could be
| Automated tests included?              | no (only JS changes, currently no tests infrastructure available)
| Related user documentation PR URL      | no
| Related developer documentation PR URL | there are many comments directly in the code
| Issue(s) addressed                     | working towards enabling dynamic content for MJML fixes #10142 #9993

#### Description:
Background: There are two repositories for Mautic Grapesjs: Plugin and Preset. It was decided that we move all the general Mautic functionality in the Preset (core functionality, library). The plugin itself is only the implementation of that. It should contain only a minimum of code. Only the code necessary for wiring Grapesjs into Mautic (similar to a controller). This PR takes the first step in that direction.
- moving code from plugin to preset
- using classes and separate files for better code maintainability
- importing external libraries via npm and JS: `import`

#### No more known Issue => fixed
~There is a bug with the dependency on CKEditor. So the dynamic content (textareas) can not be edited. Until this is resolved, testing is only possible for general functionality.~

fixes: https://github.com/mautic/mautic/issues/10142
fixes: https://github.com/mautic/mautic/issues/9997
fixes: https://github.com/mautic/mautic/issues/10246
fixes: https://github.com/mautic/mautic/issues/10000

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a new dynamic content block
3. Send the email and verify that the correct block is displayed.

**Please test the steps above both with the new GrapesJS builder as well as the legacy builder**.
